File: src/coordination/AsyncAutoResetEvent.ts
// @leosingleton/commonlibs - Common Libraries for TypeScript and .NET Core
// Copyright (c) Leo C. Singleton IV <leo@leosingleton.com>
// See LICENSE in the project root for license information.

import { AsyncEventWaitHandle } from './AsyncEventWaitHandle';

/**
 * Async version of .NET's System.Threading.AutoResetEvent
 */
export class AsyncAutoResetEvent extends AsyncEventWaitHandle {
  public constructor(initialState = false) {
    super(true, initialState);
  }
}

File: dotnet/CommonLibs/Coordination/AsyncAutoResetEvent.cs
// LeoSingleton.CommonLibs - Common Libraries for TypeScript and .NET Core
// Copyright (c) Leo C. Singleton IV <leo@leosingleton.com>
// See LICENSE in the project root for license information.

namespace LeoSingleton.CommonLibs.Coordination
{
    /// <summary>
    /// Async version of <see cref="System.Threading.AutoResetEvent"/>
    /// </summary>
    public class AsyncAutoResetEvent : AsyncEventWaitHandle
    {
        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="initialState">Whether the event is intially set</param>
        public AsyncAutoResetEvent(bool initialState = false) : base(true, initialState)
        {
        }
    }
}
